OCPBUGS#19055: cert-manager: Configure CPU and Memory (#64779)
openshift/openshift-docs: modules/cert-manager-configure-cpu-memory.adoc

  metadata:
    name: cert-manager-cainjector
    namespace: cert-manager
...
  spec:
    template:
      spec:
        containers:
        - name: cert-manager-cainjector
          resources: {} <1>
...
  metadata:
    name: cert-manager-webhook
    namespace: cert-manager
...
  spec:
    template:
      spec:
        containers:
        - name: cert-manager-webhook
          resources: {} <1>
...
----
<1> The empty `spec.resources` field indicates that by default, the cert-manager components do not have CPU and memory limits.

. To configure the CPU and memory limits for the cert-manager controller, CA injector, and Webhook, enter the following command:
+
[source,terminal]
----
$ oc patch certmanager.operator cluster --type=merge -p="
spec:
  controllerConfig:
    overrideResources:
      limits:
        cpu: 200m
        memory: 64Mi
      requests:
        cpu: 10m
        memory: 16Mi
  webhookConfig:
    overrideResources:
      limits:
        cpu: 200m
        memory: 64Mi
      requests:
        cpu: 10m
        memory: 16Mi
  cainjectorConfig:
    overrideResources:
      limits:
        cpu: 200m
        memory: 64Mi
      requests:
        cpu: 10m
        memory: 16Mi

> "

Reviewer: Can you please update the `*m`, `*Mi` values and add placeholders for them?
i.e. let these values be user-customizable and add <1>, <2>, <3>, ... descriptions for each please, similar to what is done below in case of operator limit setting.

// Module included in the following assemblies:
//
// * security/cert_manager_operator/cert-manager-customizing-api-fields.adoc

:_content-type: PROCEDURE
[id="cert-manager-configure-cpu-memory_{context}"]
= Overriding CPU and memory limits for the cert-manager components

> After installing the {cert-manager-operator}, you can configure the CPU and memory limits from the cert-manager Operator API for the cert-manager components such as cert-manager controller, CA injector, and Webhook.

Reviewer 2: I think cert-manager Operator doesnt need backticks but I see there is an attribute so does it need to be used in the instance before API? AlsoI think components and controller do need backticks, so 
```suggestion
After installing the {cert-manager-operator}, you can configure the CPU and memory limits from the {cert-manager-operator} API for the `cert-manager` components such as `cert-manager` controller, CA injector, and Webhook.
```

Author: IMO "cert-manager" would not be in backticks. In PascalCase typically it is --> case  dependent:"`CertManager`", case independent:"`certmanager`"- which we use in backticks.

.Prerequisites

* You have access to the {product-title} cluster as a user with the `cluster-admin` role.
* You have installed the {cert-manager-operator} 1.12.0 or later.

.Procedure

> . Check that the deployments of the cert-manager controller, CA injector, and Webhook are available by entering the following command:

Reviewer 2: Would cert-manager be in backticks here too?

.Example output
[source,terminal]
----
NAME                      READY   UP-TO-DATE   AVAILABLE   AGE
cert-manager              1/1     1            1           53m
cert-manager-cainjector   1/1     1            1           53m
cert-manager-webhook      1/1     1            1           53m
----

. Before setting the CPU and memory limit, check the existing configuration for the cert-manager controller, CA injector, and Webhook by entering the following command:
+
[source,terminal]
----
$ oc get deployment -n cert-manager -o yaml
----
+
.Example output
[source,terminal]
----
# ...
  metadata:
    name: cert-manager
    namespace: cert-manager
# ...
  spec:
    template:
      spec:
        containers:
        - name: cert-manager-controller
          resources: {} <1>
# ...
  metadata:
    name: cert-manager-cainjector
    namespace: cert-manager
# ...
  spec:
    template:
      spec:
        containers:
        - name: cert-manager-cainjector
          resources: {} <1>
# ...
  metadata:
    name: cert-manager-webhook
    namespace: cert-manager
# ...
  spec:
    template:
      spec:
        containers:
        - name: cert-manager-webhook
          resources: {} <1>
# ...
----

> <1> The empty `spec.resources` field indicates that by default, the cert-manager components do not have CPU and memory limits.

Reviewer 2: ```suggestion
<1> The `spec.resources` field is empty by default. The cert-manager components do not have CPU and memory limits.
```

> . To configure the CPU and memory limits for the cert-manager controller, CA injector, and Webhook, enter the following command:

Reviewer 2: I would veryify thoughout this PR if cert-manager should be in backticks.
